compute_tools/src/migration.rs

```diff
@@ -65,10 +66,35 @@ impl<'m> MigrationRunner<'m> {
                 };
             }
 
+            // We use this fail point in order to check that failing in the
+            // middle of applying a series of migrations fails in an expected
+            // manner
+            {
```

Comment:
I think failpoint should be after you executed migration and advanced the migration id, BUT before commit. Otherwise, it doesn't make much sense -- you fail before even trying to run a migration

So I suggest moving it before COMMIT

Reply:
Without it being here, I would need two fail points. One in the skip block, and another in the apply block, which I can do, but just want to make sure you understood why I put it where it is

Reply:
This is also kind of why I made my comment about adding an additional test.

Reply:
Can you elaborate why we need another failpoint at this stage?

Current migrations are transactional, so it doesn't matter at which stage of it we will fail, right?

Right now, you fail even before BEGIN, if we fail before COMMIT, it should give exactly the same result. There are two code paths, though, but I think we can put a failpoint inside `update_migration_id` (maybe together with renaming the failpoint slightly to make it more clear), which is used in both. Or do I miss something?